Source/WebCore/page/Frame.cpp

```diff
@@ -257,4 +258,27 @@ void Frame::setOwnerElement(HTMLFrameOwnerElement* element)
     }
 }
 
+void Frame::setOwnerPermissionsPolicy(OwnerPermissionsPolicyData&& ownerPermissionsPolicy)
+{
+    // If the frame is in the same process as parent, it should compute policy from owner element directly.
+    if (dynamicDowncast<RemoteFrame>(tree().parent()))
+        m_ownerPermisssionsPolicy = WTFMove(ownerPermissionsPolicy);
+}
+
+std::optional<OwnerPermissionsPolicyData> Frame::ownerPermissionsPolicy()
+{
+    if (m_ownerPermisssionsPolicy)
+        return m_ownerPermisssionsPolicy;
+
+    RefPtr owner = ownerElement();
+    if (!owner)
+        return std::nullopt;
+
+    auto documentOrigin = WebCore::SecurityOriginData { owner->document().securityOrigin().data() };
+    auto documentPolicy = owner->document().permissionsPolicy();
+    RefPtr iframe = dynamicDowncast<HTMLIFrameElement>(owner);
+    auto containerPolicy = iframe ? iframe->permissionsPolicyDirective() : WebCore::PermissionsPolicy::PolicyDirective { };
+    return OwnerPermissionsPolicyData { WTFMove(documentOrigin), WTFMove(documentPolicy), WTFMove(containerPolicy) };
```

Comment:
Could we set m_ownerPermisssionsPolicy to this value?

Reply:
We probably should not cache m_ownerPermisssionsPolicy because frame that resides in the same process as its parent should be able to get the latest policy value by computation. `m_ownerPermisssionsPolicy` should only be used for frame that cannot access its parent. 
I will rename the parameters to make it more clear.